src/setupTests.ts

```diff
@@ -3,3 +3,37 @@
 // expect(element).toHaveTextContent(/react/i)
 // learn more: https://github.com/testing-library/jest-dom
 import "@testing-library/jest-dom/extend-expect";
+
```

Comment:
Personally, I'm indifferent. Some projects like to put them nearest to what they test so its easy to find the relevant tests. Others like the more traditional testing folder with unit / integration / system tests. I think both are fine.

 This is from the [Create React App Docs](https://create-react-app.dev/docs/running-tests#filename-conventions):

> Jest will look for test files with any of the following popular naming conventions:
> 
> Files with .js suffix in __tests__ folders.
> Files with .test.js suffix.
> Files with .spec.js suffix.
> The .test.js / .spec.js files (or the __tests__ folders) can be located at any depth under the src top level folder.
> 
> We recommend to put the test files (or __tests__ folders) next to the code they are testing so that relative imports appear shorter. For example, if App.test.js and App.js are in the same folder, the test only needs to import App from './App' instead of a long relative path. Collocation also helps find tests more quickly in larger projects.

Any preference?

Reply:
the relative paths stuff definitely resonates with me (memories of ../../../../../.. come back...) - I'm happy with this as is, as its a good point. I wanted to suggest a tests folder wrapping them, but we'd presumably want the tests to follow a file structure like the source itself, so that'd create the same problem it'd be trying to solve